packages/web3-utils/src/formatter.ts

```diff
@@ -174,7 +175,29 @@ export const convert = (
 
 		// If value is an array
 		if (Array.isArray(value)) {
-			if (isNullish(schemaProp?.items)) {
+			let _schemaProp = schemaProp;
+
+			// TODO This is a naive approach to solving the issue of
+			// a schema using oneOf. This chunk of code was intended to handle
+			// BlockSchema.transactions
+			// TODO BlockSchema.transactions are not being formatted
+			if (schemaProp?.oneOf) {
+				for (const oneOfSchemaProp of schemaProp.oneOf) {
+					if (
+						!Array.isArray(schemaProp?.items) &&
+						((typeof value[0] === 'object' &&
+							((oneOfSchemaProp as JsonSchema)?.items as JsonSchema)?.type ===
+								'object') ||
+							(typeof value[0] === 'string' &&
+								((oneOfSchemaProp as JsonSchema)?.items as JsonSchema)?.type !==
+									'object'))
```

Comment:
This is the fix that's specific to `blockSchema.transactions`, and needs to be refactored to be more general.
It's checking if the first value in the `transactions` array is a hash or a transaction object and uses the corresponding schema

Reply:
Created #5370 to track this